packages/app/pages/checkout/[network]/[address].tsx

```diff
@@ -133,6 +157,18 @@ const CheckoutPage = ({
   const [error, setError] = useState("");
   const [loading, setLoading] = useState(false);
   const [supply, setSupply] = useState(initialSupply);
+  const [url, setUrl] = useState("");
+  useEffect(() => {
+    // Fetch redirect url from DB on page load.
+    (async function() {
+      try {
+          const response = await getRedirectionUrl(db, address);
+          setUrl(response);
+      } catch (e) {
+          console.error(e);
+      }
+    })();
+  }, []);
```

Comment:
we need to add `setUrl` in the dependencies here

Reply:
done